cardano/src/block/date.rs

```diff
@@ -65,6 +65,15 @@ impl BlockDate {
             &BlockDate::Normal(ref s) => s.epoch,
         }
     }
+    pub fn slotid(&self) -> Option<SlotId> {
+        match self {
+            &BlockDate::Boundary(_) => None,
+            &BlockDate::Normal(ref s) => Some(s.slotid),
+        }
+    }
+    pub fn epoch_and_slot(&self) -> (EpochId, Option<SlotId>) {
+        (self.get_epochid(), self.slotid())
+    }
```

Comment:
Just sugar functions to get epoch number and optional slot number from a block date